project/SamplesCompilationProject.scala

```diff
@@ -47,7 +51,30 @@ object SamplesCompilationProject {
                 // Disable tests for composite projects since they're primarily for compilation verification
                 // Maven sets akka.javasdk.dev-mode.project-artifact-id automatically but SBT composite projects don't have this
                 Test / test := {},
-                Test / testOnly := {})
+                Test / testOnly := {},
+                // Only run prettier-maven-plugin if there are sources to compile
+                formatIfNeeded := {
+                  val srcs = (Compile / sources).value
+                  val targetDir = (Compile / compile / streams).value.cacheDirectory
+                  val markerFile = targetDir / "prettier-last-run"
+                  val changed =
+                    if (markerFile.exists) {
+                      srcs.exists(src => src.lastModified() > markerFile.lastModified())
+                    } else {
+                      srcs.nonEmpty
```

Comment:
> One drawback is that when compile a new checkout, it will always run a formatting task. It slows down the build a bit.

How about something like the following to check a git diff 
```suggestion
              Process(Seq("git", "diff", "--name-only","."), baseDirectory.value).!!.linesIterator.map(_.trim).exists(_.nonEmpty)
```

Maybe even look at `--staged`

Reply:
Interesting. 

Using the diff can be interesting to decide when to create the `markerFile` for the first time. For example, you do a `clean compile`, if there is not diff, not worth running the formatter at all. 

But let's assume you change a file. Now you have a diff and the formatter is applied. After that, you still have a diff, but you don't need to format anymore since last change was made by the formatter. 

Maybe a combination of both will be better, but I want to give another try and see if I can have sbt doing this trick for me. Sbt does know when there are files to compile, it's just that I couldn't find yet how to get this info.